src/csexp_rpc/csexp_rpc.mli

```diff
@@ -0,0 +1,58 @@
+(** Threaded RPC Client & Server for Csexp payloads. This RPC implementation is
+    entirely structure free and it is up to users of this library to impose some
+    sort of conventions on the protocol.
+
+    On Unix/Mac, RPC uses Unix domain sockets. On Windows, named pipes are used.
+
+    All functions reurning a fiber are non-blocking and run in their own thread. *)
+open Stdune
+
+module Scheduler : sig
+  (** Hook into the fiber scheduler. *)
+  type t =
+    { on_event : Fiber.fill -> unit  (** Called when any rpc event completes *)
+    ; register_pending_ivar : unit -> unit
+          (** Called before any ivar is created *)
+    ; thread : 'a 'b. ('a -> 'b) -> 'a -> Thread.t
```

Comment:
I'd use:

```suggestion
    ; spawn_thread : (unit ->  unit) -> Thread.t
```

- `spawn_thread` is more explicit than `thread` about what it does. `create_thread` would work too
- `(unit -> unit)` is powerful enough and simpler

Reply:
Indeed. I did that. I even changed it to `(unit -> unit) -> unit` as there's really nothing we should do with `Thread.t`